JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -253,10 +253,14 @@ private fun HomeScreenErrorAndContent(
 ) {
     if (posts.data != null) {
         PostList(posts.data, navigateTo, favorites, onToggleFavorite, modifier)
-    } else if (!posts.hasError) {
-        // if there are no posts, and no error, let the user refresh manually
-        TextButton(onClick = onRefresh, modifier.fillMaxSize()) {
-            Text("Tap to load content", textAlign = TextAlign.Center)
+    } else {
+        Box(Modifier.fillMaxSize()) {
```

Comment:
Nit: Is this needed considering the `TextButton` below has the same modifier? 
Also, if `Box` is necessary here, it should use the `modifier` passed as a param instead of `TextButton`

Reply:
The flow control is fairly confusing here, let me rework it.

Reply:
Simplified it to a three-way branch instead of nested ifs. Please take a look again!